File: CzhERP.Application/Entity/StoCountItem.cs
namespace CzhERP.Application.Entity;

[Tenant("1300000000001")]
[SugarTable("Sto_CountItem", "盘点单明细表")]
public partial class StoCountItem : EntityBase
{
    [Required]
    [SugarColumn(ColumnName = "CountId", ColumnDescription = "盘点单ID")]
    public virtual long CountId { get; set; }

    [Required]
    [SugarColumn(ColumnName = "MaterialId", ColumnDescription = "物料ID")]
    public virtual long MaterialId { get; set; }

    [Required]
    [SugarColumn(ColumnName = "MaterialCode", ColumnDescription = "物料编码", Length = 50)]
    public virtual string MaterialCode { get; set; }

    [Required]
    [SugarColumn(ColumnName = "MaterialName", ColumnDescription = "物料名称", Length = 100)]
    public virtual string MaterialName { get; set; }

    [SugarColumn(ColumnName = "Spec", ColumnDescription = "规格型号", Length = 100)]
    public virtual string? Spec { get; set; }

    [Required]
    [SugarColumn(ColumnName = "Unit", ColumnDescription = "单位", Length = 20, DefaultValue = "个")]
    public virtual string Unit { get; set; } = "个";

    [SugarColumn(ColumnName = "SystemQuantity", ColumnDescription = "系统数量", DefaultValue = "0")]
    public virtual decimal SystemQuantity { get; set; } = 0;

    [SugarColumn(ColumnName = "ActualQuantity", ColumnDescription = "实际数量", DefaultValue = "0")]
    public virtual decimal ActualQuantity { get; set; } = 0;

    [SugarColumn(ColumnName = "DiffQuantity", ColumnDescription = "差异数量", DefaultValue = "0")]
    public virtual decimal DiffQuantity { get; set; } = 0;

    [SugarColumn(ColumnName = "CostPrice", ColumnDescription = "成本单价", DefaultValue = "0")]
    public virtual decimal CostPrice { get; set; } = 0;

    [SugarColumn(ColumnName = "DiffAmount", ColumnDescription = "差异金额", DefaultValue = "0")]
    public virtual decimal DiffAmount { get; set; } = 0;

    [SugarColumn(ColumnName = "BatchNo", ColumnDescription = "批号", Length = 50)]
    public virtual string? BatchNo { get; set; }

    [SugarColumn(ColumnName = "LocationCode", ColumnDescription = "库位编码", Length = 50)]
    public virtual string? LocationCode { get; set; }

    [SugarColumn(ColumnName = "SortOrder", ColumnDescription = "排序号", DefaultValue = "0")]
    public virtual int SortOrder { get; set; } = 0;

    [SugarColumn(ColumnName = "Remark", ColumnDescription = "备注", Length = 200)]
    public virtual string? Remark { get; set; }
}

File: Web/src/api/stock/stoCountItem.ts
import {useBaseApi} from '/@/api/base';

// 盘点单明细表接口服务
export const useStoCountItemApi = () => {
	const baseApi = useBaseApi("stoCountItem");
	return {
		// 分页查询盘点单明细表
		page: baseApi.page,
		// 查看盘点单明细表详细
		detail: baseApi.detail,
		// 新增盘点单明细表
		add: baseApi.add,
		// 更新盘点单明细表
		update: baseApi.update,
		// 删除盘点单明细表
		delete: baseApi.delete,
		// 批量删除盘点单明细表
		batchDelete: baseApi.batchDelete,
		// 导出盘点单明细表数据
		exportData: baseApi.exportData,
		// 导入盘点单明细表数据
		importData: baseApi.importData,
		// 下载盘点单明细表数据导入模板
		downloadTemplate: baseApi.downloadTemplate,
	}
}

// 盘点单明细表实体
export interface StoCountItem {
	// 主键Id
	id: number;
	// 盘点单ID
	countId?: number;
	// 物料ID
	materialId?: number;
	// 物料编码
	materialCode?: string;
	// 物料名称
	materialName?: string;
	// 规格型号
	spec: string;
	// 单位
	unit?: string;
	// 系统数量
	systemQuantity?: number;
	// 实际数量
	actualQuantity?: number;
	// 差异数量
	diffQuantity?: number;
	// 成本单价
	costPrice?: number;
	// 差异金额
	diffAmount?: number;
	// 批号
	batchNo: string;
	// 库位编码
	locationCode: string;
	// 排序号
	sortOrder?: number;
	// 备注
	remark: string;
	// 创建时间
	createTime: string;
	// 更新时间
	updateTime: string;
	// 创建者Id
	createUserId: number;
	// 创建者姓名
	createUserName: string;
	// 修改者Id
	updateUserId: number;
	// 修改者姓名
	updateUserName: string;
}